sharedbuffers/mapped_struct.py

```diff
@@ -494,46 +786,32 @@ def __delitem__(self, index):
         raise AttributeError("Proxy objects are read-only")
 
     def __iter__(self):
-        for i in xrange(self.objlen):
+        for i in xrange(len(self)):
             yield self[i]
 
     def __reversed__(self):
-        for i in reversed(xrange(self.objlen)):
+        for i in reversed(xrange(len(self))):
             yield self[i]
 
     def __contains__(self, item):
-        for i in xrange(self.objlen):
-            if self[i] ==  item:
+        for e in self:
+            if e ==  item:
                 return True
         return False
 
+    def __repr__(self):
+        return str(self)
+
     def __str__(self):
         return "proxied_list([%s])" % ",".join([str(x) for x in self])
```

Comment:
I'd make `str` mimic the str of a list, and `repr` add the `proxied_list` wrapper to clarify that it's not just a list.